### ISSUE TITLE
GNS3 VM crashes once a day

### ISSUE THREAD

**Issue by dkhenderson2** [GNS3/gns3-vm#165]
Community Link: https://gns3.com/gns3-vm-crashes-once-a-day
GNS3 Version: GNS3 v2.2.31

Operating System: Windows server 2019 Core

GNS3 VM is running on Hyper-V
WindowsBuildLabEx                                       : 17763.1.amd64fre.rs5_release.180914-1434
WindowsCurrentVersion                                   : 6.3
WindowsEditionId                                        : ServerStandard
WindowsInstallationType                                 : Server Core
WindowsProductName                                      : Windows Server 2019 Standard
WindowsVersion                                          : 1809
CsManufacturer                                          : Dell Inc.
CsModel                                                 : Precision WorkStation T3500
CsNetworkAdapters                                       : {Ethernet, Ethernet 2, Ethernet 3, Ethernet 4...}
CsNetworkServerModeEnabled                              : True
CsNumberOfLogicalProcessors                             : 12
CsNumberOfProcessors                                    : 1
CsProcessors                                            : {Intel(R) Xeon(R) CPU           W3690  @ 3.47GHz}
CsOEMStringArray                                        : {www.dell.com}
CsSystemType                                            : x64-based PC
OsHotFixes                                              : {KB5008873, KB4464455, KB5008287}
OsBuildType                                             : Multiprocessor Free
OsTotalVisibleMemorySize                                : 50329140
OsFreePhysicalMemory                                    : 8871120
OsTotalVirtualMemorySize                                : 57669172
OsFreeVirtualMemory                                     : 16291536
OsInUseVirtualMemory                                    : 41377636
OsTotalSwapSpaceSize                                    :
OsManufacturer                                          : Microsoft Corporation
OsArchitecture                                          : 64-bit
OsLanguage                                              : en-US
OsProductType                                           : Server
DeviceGuardAvailableSecurityProperties                  : {BaseVirtualizationSupport, DMAProtection}

I have 4 other VM's running on the host, the GNS3 VM is the only one that crashes. doesn't matter if any labs are running or not.

Once everyday The GNS3 VM crashes, the VM reports the following error:

'GNS3VM' has encountered a fatal error.  The guest operating system reported that it failed with the following error codes: ErrorCode0: 0x0, ErrorCode1: 0x0, ErrorCode2: 0x0, ErrorCode3: 0x0, ErrorCode4: 0x0.  If the problem persists, contact Product Support for the guest operating system.  (Virtual machine ID C575362A-4F63-4B66-A67B-85F2BB783399)

Guest message:

[    6.551810] virbr0: port 1(virbr0-nic) entered blocking state

[    6.551817] virbr0: port 1(virbr0-nic) entered disabled state

[    6.551956] device virbr0-nic entered promiscuous mode

[    6.588915] bpfilter: Loaded bpfilter_umh pid 866

[    6.589234] Started bpfilter

[    6.961713] virbr0: port 1(virbr0-nic) entered blocking state

[    6.961720] virbr0: port 1(virbr0-nic) entered listening state

[    7.019967] virbr0: port 1(virbr0-nic) entered disabled state

[    7.205114] Bridge firewalling registered

[    7.272637] Initializing XFRM netlink socket

[   10.862199] loop6: detected capacity change from 0 to 8

[   52.953269] hv_balloon: Max. dynamic memory size: 12000 MB

[  175.997661] L1TF CPU bug present and SMT on, data leak possible. See CVE-2018-3646 and https://www.kernel.org/doc/html/latest/admin-guide/hw-vuln/l1tf.html for details.

[  189.315387] kvm [1445]: vcpu0, guest rIP: 0xffffffff8104b3f8 disabled perfctr wrmsr: 0xc2 data 0xffff

[12845.099399] traps: PANIC: double fault, error_code: 0x0

[12845.099402] double fault: 0000 [#1] SMP PTI

[12845.099403] CPU: 0 PID: 2445 Comm: gpgconf Not tainted 5.13.0-39-generic #44~20.04.1-Ubuntu

[12845.099403] Hardware name: Microsoft Corporation Virtual Machine/Virtual Machine, BIOS 090007  05/18/2018

[12845.099404] RIP: 0010:0x15f0

[12845.099404] Code: Unable to access opcode bytes at RIP 0x15c6.

[12845.099405] RSP: 0018:0000000000003000 EFLAGS: 00010096

[12845.099406] RAX: 000000000000002d RBX: 0000000000000000 RCX: 0000000000000000

[12845.099406] RDX: 00000000f7e75000 RSI: 0000000000021000 RDI: 00000000f7e75000

[12845.099407] RBP: 00000000ffa317d8 R08: 0000000000000000 R09: 0000000000000000

[12845.099407] R10: 0000000000000000 R11: 0000000000000000 R12: 0000000000000000

[12845.099408] R13: 0000000000000000 R14: 0000000000000000 R15: 0000000000000000

[12845.099408] FS:  0000000000000000(0000) GS:ffff9c6be6200000(0063) knlGS:00000000f7f8e940

[12845.099409] CS:  0010 DS: 002b ES: 002b CR0: 0000000080050033

[12845.099409] CR2: 0000000000002ff8 CR3: 00000001060c0001 CR4: 00000000000226f0

[12845.099409] Call Trace:

[12845.099410] Modules linked in: nf_conntrack_netlink xfrm_user xfrm_algo xt_addrtype br_netfilter xt_CHECKSUM xt_MASQUERADE xt_conntrack ipt_REJECT nf_reject_ipv4 xt_tcpudp ip6table_mangle ip6table_nat iptable_mangle iptable_nat nf_nat nf_conntrack nf_defrag_ipv6 nf_defrag_ipv4 nf_tables nfnetlink ip6table_filter ip6_tables iptable_filter bpfilter bridge stp llc aufs overlay dm_multipath scsi_dh_rdac scsi_dh_emc scsi_dh_alua kvm_intel kvm input_leds serio_raw hyperv_fb hv_balloon joydev mac_hid sch_fq_codel msr drm ip_tables x_tables autofs4 btrfs blake2b_generic zstd_compress raid10 raid456 async_raid6_recov async_memcpy async_pq async_xor async_tx xor raid6_pq libcrc32c raid1 raid0 multipath linear hid_generic crct10dif_pclmul hv_netvsc hyperv_keyboard hid_hyperv crc32_pclmul hid hv_storvsc hv_utils ghash_clmulni_intel scsi_transport_fc aesni_intel crypto_simd cryptd psmouse i2c_piix4 pata_acpi hv_vmbus floppy

[12845.099433] ---[ end trace ecb923a2bb38b496 ]---

[12845.099433] RIP: 0010:0x15f0

[12845.099434] Code: Unable to access opcode bytes at RIP 0x15c6.

[12845.099434] RSP: 0018:0000000000003000 EFLAGS: 00010096

[12845.099435] RAX: 000000000000002d RBX: 0000000000000000 RCX: 0000000000000000

[12845.099435] RDX: 00000000f7e75000 RSI: 0000000000021000 RDI: 00000000f7e75000

[12845.099436] RBP: 00000000ffa317d8 R08: 0000000000000000 R09: 0000000000000000

[12845.099436] R10: 0000000000000000 R11: 0000000000000000 R12: 0000000000000000

[12845.099437] R13: 0000000000000000 R14: 0000000000000000 R15: 0000000000000000

[12845.099437] FS:  0000000000000000(0000) GS:ffff9c6be6200000(0063) knlGS:00000000f7f8e940

[12845.099438] CS:  0010 DS: 002b ES: 002b CR0: 0000000080050033

[12845.099438] CR2: 0000000000002ff8 CR3: 00000001060c0001 CR4: 00000000000226f0

[12845.099439] Kernel panic - not syncing: Fatal exception in interrupt

[12845.099439] Kernel Offset: 0x29800000 from 0xffffffff81000000 (relocation range: 0xffffffff80000000-0xffffffffbfffffff)

**Comment by grossmj** [comment]
Please try this new version of the GNS3 VM based on the latest Ubuntu 22.04 LTS: https://we.tl/t-u8O9zLeagx

Note that it is very new and it hasn't been extensively tested.

Let me know how stable the VM is, thanks :)

**Comment by dkhenderson2** [comment]
link isn't working

**Comment by grossmj** [comment]
Hum the link is working here. Maybe you are trying to download from a restricted country?

![Screenshot from 2022-04-28 12-34-02](https://user-images.githubusercontent.com/4752614/165683321-7db9e99f-8579-4fa8-a7d7-15b2f8f7bf1b.png)

**Comment by dkhenderson2** [comment]
Got it, I never used Wetransfer before, the giant ad threw me off and didn't notice the small download link.

Ill test this build out and let you know how it preforms.

**Comment by dkhenderson2** [comment]
Running smooth for 1 Day 6 hours. so far no issues.

**Comment by dkhenderson2** [comment]
Seeing a weird issue

1. When I try to create a new QEMU template based on vios_l2-adventerprisek9-m.03.2017.qcow2 image. The image will boot up but immediately crashes.

3. If I import a project that already have the same image(vios_l2-adventerprisek9-m.03.2017.qcow2) in use, when I start the project the image boots and running normally.

**Comment by dkhenderson2** [comment]
I deleted and rebuilt the VM from the new v2.2.32 that was released. It runs the vios_l2 image with no issue. But still has the same symptoms where it reboots randomly with in 24hrs of powering on VM.

I redownloaded the v2.2.31 you provided and still getting the same issue running vios_l2 image error below.

alidblock_diagnose, code = 1

current memory block, bp = 0x14D1BCB0,
memorypool type is Processor
data check, ptr = 0x14D1BCD8

next memory block, bp = 0x14D1BCF8,
memorypool type is Processor
data check, ptr = 0x14D1BD20

previous memory block, bp = 0x14D1B968,
memorypool type is Processor
data check, ptr = 0x14D1B990
========= Dump bp = 0x14D1BCB0 ======================

14D1BBB0:        0        0        0        0        0        0        0        0
14D1BBD0:        0        0        0        0        0        0        0        0
14D1BBF0:        0        0        0        0        0        0        0        0
14D1BC10:        0        0        0        0        0        0        0        0
14D1BC30:        0        0        0        0        0        0        0        0
14D1BC50:        0        0        0        0        0        0        0        0
14D1BC70:        0        0        0        0        0        0        0        0
14D1BC90:        0        0        0        0        0        0        0 FD0110DF
14D1BCB0: AB1234CD   300000       30  412E2A4   90813B 14D1BCF8 14D1B97C 80000010
14D1BCD0:        1  14CCF15  1000000        0 5D780300        0        1 BAAE0000
14D1BCF0: 1529FF0C 1529FF0C        0        0        0        0        0        0
14D1BD10:        0        0        0        0        0        0        0        0
14D1BD30:        0        0        0        0        0        0        0        0
14D1BD50:        0        0        0        0        0        0        0        0
14D1BD70:        0        0        0        0        0        0        0        0
14D1BD90:        0        0        0        0        0        0        0        0
========= Dump bp->next = 0x14D1BCF8 ======================

14D1BBF8:        0        0        0        0        0        0        0        0
14D1BC18:        0        0        0        0        0        0        0        0
14D1BC38:        0        0        0        0        0        0        0        0
14D1BC58:        0        0        0        0        0        0        0        0
14D1BC78:        0        0        0        0        0        0        0        0
14D1BC98:        0        0        0        0        0 FD0110DF AB1234CD   300000
14D1BCB8:       30  412E2A4   90813B 14D1BCF8 14D1B97C 80000010        1  14CCF15
14D1BCD8:  1000000        0 5D780300        0        1 BAAE0000 1529FF0C 1529FF0C
14D1BCF8:        0        0        0        0        0        0        0        0
14D1BD18:        0        0        0        0        0        0        0        0
14D1BD38:        0        0        0        0        0        0        0        0
14D1BD58:        0        0        0        0        0        0        0        0
14D1BD78:        0        0        0        0        0        0        0        0
14D1BD98:        0        0        0        0        0        0        0        0
14D1BDB8:        0        0        0        0        0        0        0        0
14D1BDD8:        0        0        0        0        0        0        0        0
========== Dump bp->previous = 0x14D1B97C =====================

14D1B87C: 80000010        1        0 434F5245 2D534D2D 434C4945 4E540000        0
14D1B89C:        0        0 FD0110DF AB1234CD FFFE0000        0  44C3468   947E64
14D1B8BC: 14D1B908 14D1B874 8000001C        1        0  4C1FA20        0        0
14D1B8DC:        0        0        0        0        0        0        0        0
14D1B8FC: 14D1BEB8        0 FD0110DF AB1234CD FFFE0000        0  4C1FA20   3E67A8
14D1B91C: 14D1B968 14D1B8BC 8000001C        1        0        0        5        A
14D1B93C:       2C       30        0        0        0        1        0        0
14D1B95C:        0 14D1B990 FD0110DF AB1234CD FFFE0000        0  4C1FA20   3E67A8
14D1B97C: 14D1BCB0 14D1B91C 80000190        1        0        0      391        0
14D1B99C:        0        0        0        0        0        0        0        0
14D1B9BC:        0        0        0       2C        B        1        0        0
14D1B9DC:      5BF        0      245        0        0        3        0        0
14D1B9FC:        0        0        0        0        0        0        0        2
14D1BA1C:        0        0      5BF        0      246        0        0        1
14D1BA3C:        0        0        0        0        0        0        0        0
14D1BA5C:        0        3        0        0      5BF        0      247        0
============================================

*May  4 15:24:44.010: %SYS-3-OVERRUN: Block overrun at 14D1BCB0 (red zone 1529FF0C)
-Traceback= 1DDCD0Az 34AA072z 349D9AEz 349D86Dz 3E0AD21z 907BD5z 9086C3z 908589z 908B79z 8D7BF8z 887061z 8BAE73z 8B9FD7z 8B7827z 8BCCC4z 8C0587z
*May  4 15:24:44.012: %SYS-6-MTRACE: mallocfree: addr, pc
 19D97430,1D95E54  19D97430,2000035C 1A082640,1D95E1C  1A082640,2000035C
 14D1BCD8,907BD5   15E5D5B8,34BDD50  15E5D5B8,20000084 1A0D7E78,60000010
*May  4 15:24:44.012: %SYS-6-MTRACE: mallocfree: addr, pc
 1A0D7D90,34BDF9C  1A0D7D90,34BDD50  1A0D7D90,20000084 18CF2BF8,60000010
 18CF2B10,34BDF9C  14D1BCD8,90813B   14D1BCD8,20000060 18CF2B10,34BDD50 
*May  4 15:24:44.012: %SYS-6-BLKINFO: Corrupted redzone blk 14D1BCB0, words 16, alloc 90813B, InUse, dealloc 0, rfcnt 1
-Traceback= 1DDCD0Az 349FB60z 34AA08Cz 349D9AEz 349D86Dz 3E0AD21z 907BD5z 9086C3z 908589z 908B79z 8D7BF8z 887061z 8BAE73z 8B9FD7z 8B7827z 8BCCC4z
*May  4 15:24:44.013: %SYS-6-MEMDUMP: 0x14D1BCB0: 0xAB1234CD 0x300000 0x30 0x412E2A4
*May  4 15:24:44.013: %SYS-6-MEMDUMP: 0x14D1BCC0: 0x90813B 0x14D1BCF8 0x14D1B97C 0x80000010
*May  4 15:24:44.013: %SYS-6-MEMDUMP: 0x14D1BCD0: 0x1 0x14CCF15 0x0 0x0
CPU exception occurred: Overflow(4) (SIGSTOP, 4, 0x4)


 15:24:45 UTC Wed May 4 2022: Unexpected exception to CPU: vector 4, PC/EIP = 0x8E18E9  

-Traceback= 8E18E9 88270C 34AA096 349D9AE 349D86D 3E0AD21 907BD5 9086C3 908589 908B79 8D7BF8 887061 8BAE73 8B9FD7 8B7827 8BCCC4

CPU Register Context:
EAX = 0x00000000  ECX     = 0x000052D2  EDX = 0x141D25F0  EBX  = 0x00000000
ESP = 0x153C2EBC  EBP     = 0x153C2ED4  ESI = 0xB0BCD114  EDI  = 0x2027E204
EIP = 0x008E18E9  EFLAGS  = 0x00000246  CS  = 0x00000008  SS   = 0x00000010
DS  = 0x00000010  ES      = 0x00000010  FS  = 0x00000010  GS   = 0x00000010

Writing crashinfo to flash0:crashinfo_20220504-152445-UTC
